### PR TITLE
Properly close GRPC ManagedChannel

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Properly close GRPC managedChannel. 
For HTTP Client, close does not do anything as OkHttpClient does not need to be shutdown separately.
https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#shutdown-isnt-necessary

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #305 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/binding/http/BindingIT.java [hunk 1]
```diff
@@ -123,5 +123,6 @@ public void inputOutputBinding() throws Exception {
       }
       assertEquals("hello", resultClass.message);
     }, 8000);
+    client.close();
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/binding/http/BindingIT.java:126]: Use try-with-resources instead of closing it explicitly.

Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/binding/http/BindingIT.java:126]: Change it everywhere.

Comment (mukundansundar) [reply sdk-tests/src/test/java/io/dapr/it/binding/http/BindingIT.java]: done

File: examples/src/main/java/io/dapr/examples/bindings/http/README.md [hunk 2]
```diff
@@ -110,7 +110,7 @@ public class OutputBindingExample {
   static final String BINDING_OPERATION = "create";
 ///...
   public static void main(String[] args) {
-    DaprClient client = new DaprClientBuilder().build();
+    try (DaprClient client = new DaprClientBuilder().build()) {
     ///...
     MyClass myClass = new MyClass();
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/bindings/http/README.md:115]: concatenate.

File: examples/src/main/java/io/dapr/examples/invoke/grpc/README.md [hunk 3]
```diff
@@ -87,26 +87,29 @@ The other component is the client. It will send one message per second to the se
...
+        int count = 0;
+        while (true) {
+          String message = "Message #" + (count++);
+          System.out.println("Sending message: " + message);
+          client.invokeService(Verb.POST, serviceAppId, method, message).block();
+          System.out.println("Message sent: " + message);
+    
+          Thread.sleep(1000);
+        }
+    } catch (IOException e) {
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/invoke/grpc/README.md:104]: Add this to throws and remove the catch block.

File: examples/src/main/java/io/dapr/examples/invoke/grpc/HelloWorldClient.java [hunk 4]
```diff
@@ -24,23 +26,26 @@
    * @param args Array of messages to be sent.
    */
   public static void main(String[] args) throws InterruptedException {
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/invoke/grpc/HelloWorldClient.java:28]: Copy and paste the whole method. The signature does not match.

File: examples/src/main/java/io/dapr/examples/pubsub/http/Publisher.java [hunk 5]
```diff
@@ -32,31 +33,34 @@
...
+      System.out.println("Published one byte.");
 
-    // This is an example, so for simplicity we are just exiting here.  
-    // Normally a dapr app would be a web service and not exit main.    
-    System.out.println("Done.");
+      // This is an example, so for simplicity we are just exiting here.
+      // Normally a dapr app would be a web service and not exit main.
+      System.out.println("Done.");
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/pubsub/http/Publisher.java:63]: Method has throws Exception, this should work without this catch.

File: examples/src/main/java/io/dapr/examples/pubsub/http/README.md [hunk 6]
```diff
@@ -84,19 +84,22 @@ dapr run --components-path ./components --app-id subscriber --app-port 3000 --po
...
-      System.out.println("Published message: " + message);
-      //..
+    try (DaprClient client = new DaprClientBuilder().build()) {
+        for (int i = 0; i < NUM_MESSAGES; i++) {
+          String message = String.format("This is message #%d", i);
+          client.publishEvent(TOPIC_NAME, message).block();
+          System.out.println("Published message: " + message);
+          //..
+        }
+    } catch (IOException e) {
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/pubsub/http/README.md:101]: Remove catch.

File: examples/src/main/java/io/dapr/examples/secrets/README.md [hunk 7]
```diff
@@ -78,24 +80,27 @@ The example's main function is in `SecretClient.java`.
...
     ///...
     String secretKey = args[0];
-    DaprClient client = (new DaprClientBuilder()).build();
-    Map<String, String> secret = client.getSecret(SECRET_STORE_NAME, secretKey).block();
-    System.out.println(JSON_SERIALIZER.writeValueAsString(secret));
+    try (DaprClient client = (new DaprClientBuilder()).build()) {
+        Map<String, String> secret = client.getSecret(SECRET_STORE_NAME, secretKey).block();
+        System.out.println(JSON_SERIALIZER.writeValueAsString(secret));
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/secrets/README.md:94]: Remove catch.

File: examples/src/main/java/io/dapr/examples/secrets/SecretClient.java [hunk 8]
```diff
@@ -44,8 +45,11 @@ public static void main(String[] args) throws Exception {
...
 
     String secretKey = args[0];
-    DaprClient client = (new DaprClientBuilder()).build();
-    Map<String, String> secret = client.getSecret(SECRET_STORE_NAME, secretKey).block();
-    System.out.println(JSON_SERIALIZER.writeValueAsString(secret));
+    try (DaprClient client = (new DaprClientBuilder()).build()) {
+      Map<String, String> secret = client.getSecret(SECRET_STORE_NAME, secretKey).block();
+      System.out.println(JSON_SERIALIZER.writeValueAsString(secret));
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/secrets/SecretClient.java:52]: Remove catch.

File: examples/src/main/java/io/dapr/examples/state/README.md [hunk 9]
```diff
@@ -32,36 +32,39 @@ cd examples
...
+        Mono<State<MyClass>> retrievedMessageMono = client.getState(STATE_STORE_NAME, KEY_NAME, MyClass.class);
+        System.out.println("Retrieved class message from state: " + (retrievedMessageMono.block().getValue()).message);
+    
+        System.out.println("Deleting state...");
+        Mono<Void> mono = client.deleteState(STATE_STORE_NAME, KEY_NAME);
+        mono.block();
+    
+        Mono<State<MyClass>> retrievedDeletedMessageMono = client.getState(STATE_STORE_NAME, KEY_NAME, MyClass.class);
+        System.out.println("Trying to retrieve deleted state: " + retrievedDeletedMessageMono.block().getValue());
+    } catch (IOException e) {
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/state/README.md:61]: Remove catch.

File: examples/src/main/java/io/dapr/examples/state/StateClient.java [hunk 10]
```diff
@@ -33,27 +35,30 @@
...
+      Mono<State<MyClass>> retrievedDeletedMessageMono = client.getState(STATE_STORE_NAME, KEY_NAME, MyClass.class);
+      System.out.println("Trying to retrieve deleted state: " + retrievedDeletedMessageMono.block().getValue());
 
-    // This is an example, so for simplicity we are just exiting here.  
-    // Normally a dapr app would be a web service and not exit main. 
-    System.out.println("Done");
+      // This is an example, so for simplicity we are just exiting here.
+      // Normally a dapr app would be a web service and not exit main.
+      System.out.println("Done");
+    } catch (IOException e) {
```
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/state/StateClient.java:60]: Remove catch.

**Pull request merged**